lib/pages/chat_list/chat_list.dart

```diff
@@ -159,12 +168,107 @@ class ChatListController extends State<ChatList>
     }
   }
 
+  bool hideBotsRoomFilter(Room room) {
+    return !excludedUserIds.contains(room.directChatMatrixID) &&
+        !isGroupWithOnlyBotAndUser(room);
+  }
+
   List<Room> get filteredRooms => Matrix.of(context)
       .client
       .rooms
       .where(getRoomFilterByActiveFilter(activeFilter))
+      .where(hideBotsRoomFilter)
       .toList();
 
+  String getHostName() {
+    final client = Matrix.of(context).client;
+    final userId = client.userID;
+    final userHostName = extractHostName(userId!.split(':')[1]);
+    return userHostName;
+  }
+
+  List<String> getBotIds() {
+    final hostName = getHostName();
+    return SocialNetworkManager.socialNetworks
+        .map((sn) => sn.chatBot + hostName)
+        .toList();
+  }
+
+  bool isGroupWithOnlyBotAndUser(Room room) {
+    final client = Matrix.of(context).client;
+
+    // Check that participants are fully charged
+    if (!room.participantListComplete && !loadingRooms.contains(room.id)) {
+      loadingRooms.add(room.id);
+      room.requestParticipants().then((_) {
+        loadingRooms.remove(room.id);
+        setState(() {}); // Force a rebuild to recheck the room
```

Comment:
`setState` is still potentially called a high number of times ! I thought we agreed to only rebuild once after all the Futures are resolved (which is also done)